docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -2,4 +2,173 @@
 Developer Setup
 ***************
 
-TBD
+Creating a development shell
+============================
+
+To create an interactive development ``bash`` shell:
+
+
+1. Download, install, and activate the latest `Miniforge3`_ for your system. If an existing conda (Miniforge, Miniconda, Anaconda, etc.) installation is available and writable, you may activate that and skip the first 5 recipe steps below.
```

Comment:
I know the text isn't new, but maybe
> ...the first 5 recipe steps below.
if confusing, as it seems like it might refer to this numbered list, while in fact I think it refers to the statements in the code box further down. I think this section at the top is kind of a synopsis / description of what appears in the code box, but I feel like that's not very clear. If you agree, could you see if you could make it more explicit? We might say something like "The sample code below shows the developer-setup procedure, which can be summarized as..." and then the numbered list; or maybe the code box should go first, followed by something like "The above code shows an example developer-setup procedure, which can be summarized as:" or similar.